src/execution_plan/ops/op_delete.c

```diff
@@ -189,50 +186,76 @@ static Record DeleteConsume(OpBase *opBase) {
 
 			SIValue_Free(value);
 		} else if(!(type & T_NULL)) {
-			/* Expression evaluated to a non-graph entity type
-			 * clear pending deletions and raise an exception. */
-			array_clear(op->deleted_nodes);
-			array_clear(op->deleted_edges);
-			// If evaluating the expression allocated any memory, free it.
+			// if evaluating the expression allocated any memory, free it.
 			SIValue_Free(value);
```

Comment:
I think this is also problematic, as we're passing this record onward in the `handoff` stage, an operation might try to access a freed path.